src/esn.calendar.libs/app/components/freebusy/freebusy.service.spec.js

```diff
@@ -383,32 +395,41 @@ describe('The calFreebusyService service', function() {
         attendees.push(externalAttendee);
       });
 
-      it('should call the bulk API for internal attendees only', function() {
+      it('should call the bulk API for internal attendees only', function(done) {
         getBulkFreebusyStatusStub.returns($q.when({}));
         calFreebusyService.setBulkFreeBusyStatus(attendees, start, end, [event]);
         $rootScope.$digest();
 
-        expect(getBulkFreebusyStatusStub).to.have.been.calledWith(['1', '2'], start, end, [event.uid]);
+        setTimeout(() => {
+          expect(getBulkFreebusyStatusStub).to.have.been.calledWith(['1', '2'], start, end, [event.uid]);
+          done();
+        }, 0);
       });
 
-      it('should set freebusy to unknow for external attendees', function() {
+      it('should set freebusy to unknow for external attendees', function(done) {
         getBulkFreebusyStatusStub.returns($q.when({}));
         calFreebusyService.setBulkFreeBusyStatus(attendees, start, end, [event]);
         $rootScope.$digest();
 
-        expect(externalAttendee.freeBusy).to.equal(CAL_FREEBUSY.UNKNOWN);
+        setTimeout(() => {
+          expect(externalAttendee.freeBusy).to.equal(CAL_FREEBUSY.UNKNOWN);
+          done();
+        }, 0);
       });
 
-      it('should set freebusy to unknow for internal users when freebusy status is not found', function() {
+      it('should set freebusy to unknow for internal users when freebusy status is not found', function(done) {
         getBulkFreebusyStatusStub.returns($q.when({}));
         calFreebusyService.setBulkFreeBusyStatus(attendees, start, end, [event]);
         $rootScope.$digest();
 
-        expect(attendees[0].freeBusy).to.equal(CAL_FREEBUSY.UNKNOWN);
-        expect(attendees[1].freeBusy).to.equal(CAL_FREEBUSY.UNKNOWN);
+        setTimeout(() => {
```

Comment:
Why do we need `setTimeout` here? Shouldn't `$rootScope.$digest()` do the job already?

Reply:
Normally yes. However in many cases, `$rootScope.$digest()` does not wait for the promise to be fulfilled. Therefore we have to use some `setTimeout` to ensure our test cases run after the logic code finished.

If you have any better idea please let me know. Thanks!

Reply:
It's weird that the previous code worked somehow :trollface:.

Reply:
:trollface: